ape_trezor/_cli.py

```diff
@@ -1,120 +1,152 @@
+@cli.command()
+@ape_cli_context()
+@skip_confirmation_option("Don't ask for confirmation when removing all accounts")
+def delete_all(cli_ctx, skip_confirmation):
+    """Remove all trezor accounts from your ape configuration"""
+
+    container = accounts.containers.get("trezor")
+    trezor_accounts = _get_trezor_accounts()
+    if len(trezor_accounts) == 0:
+        cli_ctx.logger.warning("No accounts found.")
+        return
 
-    path = container.data_folder.joinpath(f"{alias}.json")
-    try:
-        path.unlink()
-        notify("SUCCESS", f"Account '{alias}' has been removed")
-    except Exception as e:
-        raise Abort(f"File does not exist: {path}") from e
+    user_agrees = skip_confirmation or click.confirm("Remove all Trezor accounts from ape?")
+    if not user_agrees:
+        cli_ctx.logger.info("No account were removed.")
+        return
+
+    for account in trezor_accounts:
+        container.delete_account(account.alias)
+        cli_ctx.logger.success(f"Account '{account.alias}' has been removed.")
 
 
 @cli.command(short_help="Sign a message with your Trezor device")
 @click.argument("alias")
 @click.argument("message")
-def sign_message(alias, message):
+@ape_cli_context()
+def sign_message(cli_ctx, alias, message):
+
     if alias not in accounts.aliases:
-        notify("ERROR", f"Account with alias '{alias}' does not exist")
+        cli_ctx.logger.warning(f"Account with alias '{alias}' does not exist.")
         return
 
-    eip191message = SignableMessage(
-        version=b"E",
-        header=f"thereum Signed Message:\n{len(message)}".encode("utf8"),
-        body=message.encode("utf8"),
-    )
+    eip191message = encode_defunct(text=message)
     account = accounts.load(alias)
     signature = account.sign_message(eip191message)
-    notify("Signature:", signature.encode_vrs().hex())
+    signature_bytes = signature.encode_vrs()
+
+    # Verify signature
+    signer = Account.recover_message(eip191message, signature=signature_bytes)
+    if signer != account.address:
+        cli_ctx.abort(f"Signer resolves incorrectly, got {signer}, expected {account.address}.")
+
+    # Message signed successfully, return signature
+    click.echo(signature.encode_vrs().hex())
 
 
 @cli.command(short_help="Verify a message with your Trezor device")
 @click.argument("message")
 @click.argument("signature")
 def verify_message(message, signature):
-    eip191message = SignableMessage(
-        version=b"E",
-        header=f"thereum Signed Message:\n{len(message)}".encode("utf8"),
-        body=message.encode("utf8"),
-    )
-    notify("signer:", Account.recover_message(eip191message, signature=signature))
+
+    eip191message = encode_defunct(text=message)
+
+    try:
+        signer = Account.recover_message(eip191message, signature=signature)
+    except ValueError as exc:
+        message = "Message cannot be verified. Check the signature and try again."
+        raise TrezorSigningError(message) from exc
+
+    try:
```

Comment:
This whole `try` block can be replaced with a single line:

```
alias = accounts[signer].alias if signer in accounts else ""
```

And then on line 149, we can get rid of that `if` / `else`

Reply:
(I have tested this out)